docs/usage/pod-auto-scaling-best-practices.md

```diff
@@ -0,0 +1,332 @@
+
+HPA targets (mostly stateless) applications where adding more instances of the application can linearly increase the ability to handle additional load. It is very useful for applications that experience variable traffic patterns, as it allows for real-time scaling without the need for manual intervention.
+
+Please note: HPA continuously monitors the metrics of the targeted pods and adjusts the number of replicas based on the observed metrics. It operates solely on the current metrics when it calculates the averages across all pods, meaning it reacts to the immediate resource usage without considering past trends or patterns. Also, all pods are treated equally based on the average metrics. This could potentially lead to situations where some pods are under high load while others are underutilized. Therefore, particular care must be applied to (fair) load-balancing (connection vs. request vs. actual resource load balancing are crucial).
+
+## [Vertical Pod Autoscaling (VPA)](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler#readme)
+
+Vertical Pod Autoscaling, on the other hand, focuses on adjusting the CPU and memory resources allocated to the pods themselves. Instead of changing the number of replicas, VPA tweaks the resource requests (and limits, but only [proportionally](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#keeping-limit-proportional-to-request), if configured) for the pods in a deployment, replica set, stateful set, daemon set, or [anything really with a scale subresource](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/FAQ.md#how-can-i-apply-vpa-to-my-custom-resource). This means that each pod can be given more, or fewer resources as needed.
+
+VPA is very useful for optimizing the resource requests of pods that have dynamic resource needs over time. It does so by mutating pod requests (unfortunately, [not in-place](https://github.com/kubernetes/design-proposals-archive/blob/main/autoscaling/vertical-pod-autoscaler.md#in-place-updates)). Therefore, in order to apply new recommendations, pods that are "out of bounds" (i.e. below a configured/computed lower or above a configured/computed upper recommendation percentile) will be evicted proactively, but also pods that are "within bounds" may be evicted after a grace period. The [replication controller](https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller) will then recreate a new pod that VPA will then mutate to set the currently recommended requests (and proportional limits, if configured).
```

Comment:
Yeah, I had something similar like "higher level" and "owner resource", but I didn't like it very much as it's so unspecific.

How about I drop the link, because that's clearly wrong and we keep the "general" word "replication controller"? Would that be OK (see updated PR)? If not, I will go with your proposal, of course.

Reply:
That would be OK, let's drop the link only then.

Reply:
Thanks.